packages/react-reconciler/src/__tests__/ReactOffscreen-test.js

```diff
@@ -1620,45 +1663,50 @@ describe('ReactOffscreen', () => {
 
       nextRenderTriggerDetach = true;
 
-      // Offscreen is attached. State updates from offscreen are **not defered**.
-      // Offscreen is detached inside useLayoutEffect;
+      // Offscreen is attached and gets detached inside useLayoutEffect.
+      // State updates from offscreen are **defered**.
       await act(async () => {
         updateChildState(1);
         updateHighPriorityComponentState(1);
         expect(Scheduler).toFlushUntilNextPaint([
           'HighPriorityComponent 1',
           'Child 1',
           'HighPriorityComponent 2',
-          'Child 2',
```

Comment:
@acdlite  This PR changes how `detach` works. Detach called in an effect or in render function will be executed synchronously. Would you expect that? I am not really sure what the behaviour should be like.

Reply:
The semantic model is that it behaves exactly as if you changed the `mode` prop with a state update (other than the fact that `detach` doesn't change any styles).

Reply:
I made sure attach/detach behaves the same as if `mode` prop was changed with a state update.